skaha/src/main/java/org/opencadc/skaha/session/SessionBuilder.java

```diff
@@ -0,0 +1,307 @@
+                this.image = podContainer.getImage();
+
+                final V1ResourceRequirements resourceRequirements =
+                        Objects.requireNonNullElse(podContainer.getResources(), new V1ResourceRequirements());
+
+                final Map<String, Quantity> resourceRequests =
+                        Objects.requireNonNullElse(resourceRequirements.getRequests(), Collections.emptyMap());
+                if (this.isFixedResources) {
+                    if (resourceRequests.containsKey("memory")) {
+                        this.requestedMemory = PodResourceUsage.toCommonMemoryUnit(
+                                resourceRequests.get("memory").toSuffixedString());
+                    }
+
+                    if (resourceRequests.containsKey("cpu")) {
+                        this.requestedCPUCores = PodResourceUsage.toCoreUnit(
+                                resourceRequests.get("cpu").toSuffixedString());
+                    }
+                }
+
+                final Map<String, Quantity> resourceLimits =
+                        Objects.requireNonNullElse(resourceRequirements.getLimits(), Collections.emptyMap());
+                if (resourceLimits.containsKey("nvidia.com/gpu")) {
+                    this.requestedGPUCores =
+                            resourceLimits.get("nvidia.com/gpu").toSuffixedString();
+                } else {
+                    // Set to zero to satisfy UI conditions.
+                    this.requestedGPUCores = "0";
+                }
+            }
+
+            try {
+                this.connectURL = SessionDAO.getConnectURL(
+                        K8SUtil.getSessionsHostName(),
+                        SessionType.fromApplicationStringType(this.type),
+                        this.id,
+                        this.image,
+                        K8SUtil.getSkahaTld(),
+                        this.userID);
+            } catch (URISyntaxException e) {
+                SessionDAO.LOGGER.warn("Invalid URI for connect URL: " + this);
+                throw new IllegalStateException(e.getMessage(), e);
+            }
+        }
+
+        return this;
+    }
+
+    SessionBuilder withStatus(final V1JobStatus jobStatus, final boolean suspended) {
+        Objects.requireNonNull(jobStatus, "Invalid JobStatus");
+
+        if (jobStatus.getStartTime() != null) {
+            this.startTime = jobStatus.getStartTime().toString();
+        }
+
+        if (this.activeExpirySeconds != null && this.startTime != null) {
+            this.expiryTime = CommonUtils.getExpiryTimeString(this.startTime, this.activeExpirySeconds);
+        }
+
+        final int success = Objects.requireNonNullElse(jobStatus.getSucceeded(), 0);
+        final int failed = Objects.requireNonNullElse(jobStatus.getFailed(), 0);
+        final int active = Objects.requireNonNullElse(jobStatus.getActive(), 0);
+        final int ready = Objects.requireNonNullElse(jobStatus.getReady(), 0);
```

Comment:
Kubernetes will routinely returns jobs with no status e.g. freshly created and especially our kueue suspended jobs. 

So can listing sessions now lead to NullPointerExceptions causing the API to lie to the user?

Reply:
Perfect, I agree -- if we we know the session id exists and we dont know the status, PENDING default state works.